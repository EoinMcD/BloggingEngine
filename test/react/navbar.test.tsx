import * as React from "react";
import MyNavbar from "../../app/javascript/components/my_navbar";
import { expect, test } from "@jest/globals";
import { getByTestId, queryByText, getByText, render, RenderResult } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import "@testing-library/jest-dom";

const renderNavbar = (
  loggedIn = true,
  regPath = "/signup",
  homePath = "/"
) : RenderResult => {
  return render(
    <MyNavbar loggedIn= {loggedIn} regPath={regPath} homePath={homePath} />
  );
};

test("Component renders as expected", () => {
  const { container } = renderNavbar();
  expect(container).toMatchSnapshot();
});

test("Title button has the right link", () => {
  const { container } = renderNavbar();
  expect(getByTestId(container, "home-button")).toHaveProperty("href", window.location.href);
});

test("Register button has the right link", () => {
  const { container } = renderNavbar(false);
  expect(getByTestId(container, "reg-button")).toHaveProperty("href", window.location.href + "signup");
});

test("Register button is hidden when user is logged in", () => {
  const { container } = renderNavbar();
  expect(queryByText(container, "Register")).toBeFalsy();
});

test("Logout button is shown when user is logged in", () => {
  const { container } = renderNavbar();
  expect(getByText(container, "Logout")).toBeTruthy();
});

test("Register button is visible when user is logged out", () => {
  const { container } = renderNavbar(false);
  expect(getByText(container, "Register")).toBeTruthy();
});

test("Login button is visible when user is logged out", () => {
  const { container } = renderNavbar(false);
  expect(getByText(container, "Login")).toBeTruthy();
});
