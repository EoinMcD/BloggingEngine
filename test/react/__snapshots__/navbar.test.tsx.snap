// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders as expected 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-toolbar-4 MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h1
        class="MuiTypography-root makeStyles-title-3 MuiTypography-h1"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          data-testid="home-button"
          href="/"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            BlogEngine
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </h1>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Logout
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;
